api/provider/aws/test_aws/deploy_create_test.go

```diff
@@ -0,0 +1,112 @@
+package test_aws
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeployCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+	defer provider.SetEntityIDGenerator("dpl_id")()
+
+	// define container defaults
+	logConfig := &ecs.LogConfiguration{
+		LogDriver: aws.String("awslogs"),
+		Options: map[string]*string{
+			"awslogs-group":         aws.String("test_group"),
+			"awslogs-region":        aws.String("test_region"),
+			"awslogs-stream-prefix": aws.String("test_prefix"),
+		},
+	}
+
+	cntr1 := &ecs.ContainerDefinition{}
+	cntr1.SetName("cntr_name_1")
+	cntr1.SetLogConfiguration(logConfig)
+
+	cntr2 := &ecs.ContainerDefinition{}
+	cntr2.SetName("cntr_name_2")
+	cntr2.SetLogConfiguration(logConfig)
```

Comment:
we need at one container without the `LogConfiguration` set, so we can assert that we add the default layer0 log config correctly.

Reply:
Already called this out in a comment above but will repaste:

I'm not sure of a way to satisfy Gomock and also test whether `LogConfiguration` is injected into a given `ContainerDefinition`. Testing the `LogConfiguration` logic also seems out-of-scope for the tests in this package.